webgpu/resources/editor-settings.js

```diff
@@ -46,6 +46,25 @@ async function getText(url) {
 async function initEditor() {
   /* global monaco */
   if (typeof monaco !== 'undefined') {
+    const allLangs = monaco.languages.getLanguages();
+    const { language: jsHighligher } = await allLangs.find(({ id }) => id ==='javascript').loader();
+    // wgsl marker comment
+    jsHighligher.tokenizer.common.splice(2, 0, [
+      /(\/\*\s*[wW][gG][sS][lL]\s*\*\/)(\s*`)/, 
+      [
+        'type.identifier',
+        {
+          token: 'string',
+          next: '@wgsl_string',
+          nextEmbedded: 'wgsl'
+        }
+      ]
+    ]);
+    jsHighligher.tokenizer.wgsl_string = [
+			[/`/, { token: 'string', next: '@pop', nextEmbedded: '@pop' }],
+			[/[^\\`]+/, ''],
+		];
```

Comment:
This will break when one gets creative enough. When it encounters a backtick in `${ }`, it'll think that it's the end of the WGSL code.
```ts
let code = /*wgsl*/`struct Foo${1+``+2} { a: u32 }`;
```

I wish I knew how to fix it, but all I found is someone complaining about the Monaco documentation not quite matching what Monaco actually does https://github.com/microsoft/monaco-editor/discussions/3715